app/src/main/java/com/github/onlynotesswent/ui/search/Search.kt

```diff
@@ -38,71 +56,275 @@ import com.github.onlynotesswent.ui.overview.NoteItem
+                      author =
+                          users.value
+                              .first { it.uid == filteredNotes.value[index].userId }
+                              .userHandle()) {
+                        noteViewModel.selectedNote(filteredNotes.value[index])
+                        navigationActions.navigateTo(Screen.EDIT_NOTE)
+                      }
+                }
+              }
+        }
+        if (displayUsers) {
+          LazyColumn(
+              contentPadding = PaddingValues(horizontal = 16.dp),
+              modifier = Modifier.fillMaxWidth().padding(padding).testTag("filteredUserList")) {
+                items(filteredUsers.value.size) { index ->
+                  UserItem(filteredUsers.value[index]) {
+                    userViewModel.setProfileUser(filteredUsers.value[index])
+                    navigationActions.navigateTo(Screen.PUBLIC_PROFILE)
                   }
                 }
-          } else {
-            Text(
-                modifier =
-                    Modifier.padding(padding)
-                        .fillMaxWidth()
-                        .padding(24.dp)
-                        .testTag("noSearchResults"),
-                text = "No notes found matching your search.",
-                textAlign = TextAlign.Center,
-                color = MaterialTheme.colorScheme.onBackground)
-          }
+              }
+        }
+
+        if (displayFolders) {
+          CustomLazyGrid(
+              modifier = Modifier.fillMaxSize().padding(padding),
+              notes = remember { mutableStateOf(emptyList()) },
+              folders = filteredFolders,
+              gridModifier =
+                  Modifier.fillMaxWidth().padding(horizontal = 16.dp).testTag("filteredFolderList"),
+              folderViewModel = folderViewModel,
+              noteViewModel = noteViewModel,
+              navigationActions = navigationActions,
+              paddingValues = padding,
+              columnContent = {})
+        }
+
+        if (displayLoader) {
+          val searchedText =
+              when (searchType.value) {
+                SearchType.USERS -> "users"
+                SearchType.NOTES -> "notes"
+                SearchType.FOLDERS -> "folders"
+              }
+          Text(
+              modifier =
+                  Modifier.padding(padding)
+                      .fillMaxWidth()
+                      .padding(24.dp)
+                      .testTag("noSearchResults"),
+              text = "No $searchedText found matching your search.",
+              textAlign = TextAlign.Center,
+              color = MaterialTheme.colorScheme.onBackground)
+        }
+      }
+}
+
+@Composable
+private fun RefreshPeriodically(
+    searchQuery: MutableState<String>,
+    noteViewModel: NoteViewModel,
+    userViewModel: UserViewModel,
+    folderViewModel: FolderViewModel
+) {
+  LaunchedEffect(Unit) {
+    while (searchQuery.value.isNotBlank()) {
```

Comment:
Good function, in the future we may create a similar function that take as an input a lamda and the time to wait, that way could use in all our composable that need to fetch things periodically (like the profile picture in ProfileScreen). But good thing you added that !